### PR TITLE
Check untyped defs

### PR DESCRIPTION
**Pull request by john-kurkowski**
Catches a few more inconsistencies, and an extraneous code path.

* Fix types
* Fix docs with updated types

### CODE REVIEW

File: tldextract/cache.py [hunk 1]
```diff
@@ -219,7 +219,8 @@ def _fetch_url(session, url, timeout):
 def _make_cache_key(inputs):
     key = repr(inputs)
     try:
```
Comment (john-kurkowski) [inline tldextract/cache.py:221]: @brycedrennan do you remember what this `try` was for? According to the type checker, `repr` above returns a `str`, so `md5(key)` will always raise.

Comment (brycedrennan) [reply tldextract/cache.py:221]: It's a good question. I'm not sure. the only thing I can think of is Python 2 compatibility.

Comment (brycedrennan) [reply tldextract/cache.py:221]: yeah I see you removed python 2 compat, so that's likely what this was for

Comment (john-kurkowski) [reply tldextract/cache.py:221]: Thanks for confirming! Cool. Consider the `try` gone.

**Pull request merged**